### PR TITLE
[1LP][RFR] Added support for other providers [VM graph]

### PR DESCRIPTION
**Pull request by digitronik**
Purpose or Intent
=================
- Added support of VM graph test for other provider.
  * Azure
  * AWS
  * Vsphere6
  * RHV
- Added changes in Views as per Provider
- One test uncollected due to our infra problem; we are not getting proper metrics for EC2 disk activity. 

{{pytest: cfme/tests/candu/ -v --use-provider complete --long-running}}

### CODE REVIEW

File: cfme/tests/candu/test_vm_graph.py [hunk 1]
```diff
@@ -2,21 +2,32 @@
...
     pytest.mark.usefixtures('setup_provider'),
-    pytest.mark.provider([VMwareProvider],
+    pytest.mark.provider([VMwareProvider, RHEVMProvider, EC2Provider, AzureProvider],
                          required_fields=[(['cap_and_util', 'capandu_vm'], 'cu-24x7')])
 ]
 
 VM_GRAPHS = ['vm_cpu', 'vm_cpu_state', 'vm_memory', 'vm_disk', 'vm_network']
 
 
+# ToDo: Currently disk activity for EC2 not collecting due to infra issue.
```
Comment (jamesooden) [inline cfme/tests/candu/test_vm_graph.py]: **QUESTION:** Is there some sort of upstream bug/issue we could reference in this comment?

File: cfme/tests/candu/test_vm_graph.py [hunk 2]
```diff
@@ -43,18 +54,29 @@ def test_vm_most_recent_hour_graph_screen(graph_type, provider, enable_candu):
...
+
+    graph_data = 0
     for leg in graph.all_legends:
         graph.hide_legends(leg)
         assert not graph.legend_is_displayed(leg)
         graph.display_legends(leg)
         assert graph.legend_is_displayed(leg)
 
-        leg_data = 0
         for data in graph.data_for_legends(leg).values():
```
Comment (jamesooden) [inline cfme/tests/candu/test_vm_graph.py]: **OPTIONAL:** Could you add a comment explaining what you are trying to achieve with this part. Also, perhaps explain why the resulting sum should be greater than zero.

File: cfme/tests/candu/test_vm_graph.py (hunk 1 above)
Comment (digitronik) [reply cfme/tests/candu/test_vm_graph.py]: No, we are working on it. Its volume issue basically. 
https://forums.aws.amazon.com/thread.jspa?threadID=108613

File: cfme/infrastructure/virtual_machines.py [hunk 3]
```diff
@@ -1473,7 +1473,18 @@ def step(self, *args, **kwargs):
 
 @navigator.register(Vm, 'candu')
 class VmUtilization(CFMENavigateStep):
-    VIEW = VMUtilizationAllView
+    @property
+    def get_view(self):
```
Comment (mshriver) [inline cfme/infrastructure/virtual_machines.py]: This should just be `def VIEW(self):` property method, there's no need to wrap the method here just to change the name.

When is `VIEW=None` a valid use case that shouldn't raise an exception?

Comment (digitronik) [reply cfme/infrastructure/virtual_machines.py]: @mshriver: :+1: right, It's my bad as `CFMENavigateStep` has `VIEW=None` so I can overwrite it with the same name property.

**Comment by dajoRH** [comment]
Lint report for commit 54f4eb48084263939b1ebee6491b22b5396b6870:

`cfme/infrastructure/virtual_machines.py`:
- :red_circle: Line 1477:9: **N802** *function name should be lowercase*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by nachandr** [comment]
@digitronik , Could you add this comment?
#TODO: Add support for GCE once https://github.com/ManageIQ/integration_tests/pull/7155 is fixed

**Comment by digitronik** [comment]
@nachandr @mshriver: changes added

File: cfme/common/candu_views.py [hunk 4]
```diff
@@ -37,3 +27,32 @@ def is_displayed(self):
             )
         else:
             return "Capacity & Utilization data for Virtual Machine" in self.title.text
+
+
+class VirtualcenterVMUtilizationView(VMUtilizationView):
```
Comment (psav) [inline cfme/common/candu_views.py]: **REQUIRED**: This is a no-no I'm afraid. It's a real shame that these are not linking the integer with the particular graph. However, what we need to do, is move these Views to the provider modules. They are too specific to be here.

File: cfme/infrastructure/virtual_machines.py [hunk 5]
```diff
@@ -1473,7 +1473,15 @@ def step(self, *args, **kwargs):
 
 @navigator.register(Vm, 'candu')
 class VmUtilization(CFMENavigateStep):
-    VIEW = VMUtilizationAllView
+    @property
+    def VIEW(self):     # noqa
+        """Property returning VM Utilization Views as per provider type"""
+        if self.obj.provider.type is 'virtualcenter':
```
Comment (psav) [inline cfme/infrastructure/virtual_machines.py]: **REQUIRED**: I'm also not keen on this - what we should do instead is have a property on the provider class that links to the view instead. If the views are inside the provider modules, this actually becomes pretty easy.

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/cloud/provider/azure.py [hunk 6]
```diff
@@ -81,3 +87,7 @@ def discover_dict(credential):
             'tenant_id': getattr(credential, 'tenant_id', None),
             'subscription': getattr(credential, 'subscription_id', None)
         }
+
+    @property
```
[the remark this answers is not in the document]
Comment (digitronik) [reply cfme/cloud/provider/azure.py]: @psav: Done

**Pull request merged**